### ISSUE TITLE
Crustache JSON complier error

### ISSUE THREAD

**Issue by trans** [crystal-lang/crystal#3385]
```
require "crustache"
require "json"

data = JSON.parse("{\"this\": \"that\"}")
temp = Crustache.parse("{{this}}")

Crustache.render(temp, data.as_h)
```

**Comment by RX14** [comment]
What exactly is the error? Is this not a crustache issue?

**Comment by trans** [comment]
```
Nil assertion failed
[8560798] *raise<Exception>:NoReturn +78
[8560599] ???
[8575501] ???
[17871940] ???
[17844501] *Crystal::CodeGenVisitor#downcast<LLVM::Value, Crystal::Type+, Crystal::Type+, Bool>:LLVM::Value +1957
[18244227] *Crystal::CodeGenVisitor#visit<Crystal::Var+>:(LLVM::Value | Nil) +483
[17959614] *Crystal::ASTNode+ +222
[18139464] *Crystal::CodeGenVisitor#prepare_call_args_non_external<Crystal::Call, Crystal::Def+, Crystal::Type+>:Tuple(Array(LLVM::Value), Bool) +856
[18102357] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +389
[17960447] *Crystal::ASTNode+ +1055
[18256652] *Crystal::CodeGenVisitor#visit<Crystal::If>:Bool +140
[17961226] *Crystal::ASTNode+ +1834
[18262206] *Crystal::CodeGenVisitor#codegen_if_branch<Crystal::CodeGenVisitor::Phi, Crystal::ASTNode+, LLVM::BasicBlock, Bool>:(LLVM::Value | Nil) +62
[18257590] *Crystal::CodeGenVisitor#visit<Crystal::If>:Bool +1078
[17961226] *Crystal::ASTNode+ +1834
[18262206] *Crystal::CodeGenVisitor#codegen_if_branch<Crystal::CodeGenVisitor::Phi, Crystal::ASTNode+, LLVM::BasicBlock, Bool>:(LLVM::Value | Nil) +62
[18257560] *Crystal::CodeGenVisitor#visit<Crystal::If>:Bool +1048
[17961226] *Crystal::ASTNode+ +1834
[18028129] *Crystal::CodeGenVisitor#codegen_fun<String, Crystal::Def+, Crystal::Type+, Bool, LLVM::Module, Bool, Bool>:LLVM::Function +1889
[18053045] *Crystal::CodeGenVisitor#target_def_fun<Crystal::Def+, Crystal::Type+>:LLVM::Function +213
[18153909] *Crystal::CodeGenVisitor#codegen_call<Crystal::Call, Crystal::Def+, Crystal::Type+, Array(LLVM::Value)>:(Bool | LLVM::Value | Nil) +133
[18102582] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +614
[17960447] *Crystal::ASTNode+ +1055
[17960632] *Crystal::ASTNode+ +1240
[18028129] *Crystal::CodeGenVisitor#codegen_fun<String, Crystal::Def+, Crystal::Type+, Bool, LLVM::Module, Bool, Bool>:LLVM::Function +1889
[18053045] *Crystal::CodeGenVisitor#target_def_fun<Crystal::Def+, Crystal::Type+>:LLVM::Function +213
[18153909] *Crystal::CodeGenVisitor#codegen_call<Crystal::Call, Crystal::Def+, Crystal::Type+, Array(LLVM::Value)>:(Bool | LLVM::Value | Nil) +133
[18102582] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +614
[18107968] *Crystal::CodeGenVisitor#codegen_dispatch<Crystal::Call, Array(Crystal::Def+)>:Bool +4144
[18102039] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +71
[17960447] *Crystal::ASTNode+ +1055
[18262206] *Crystal::CodeGenVisitor#codegen_if_branch<Crystal::CodeGenVisitor::Phi, Crystal::ASTNode+, LLVM::BasicBlock, Bool>:(LLVM::Value | Nil) +62
[18257560] *Crystal::CodeGenVisitor#visit<Crystal::If>:Bool +1048
[17961226] *Crystal::ASTNode+ +1834
[18028129] *Crystal::CodeGenVisitor#codegen_fun<String, Crystal::Def+, Crystal::Type+, Bool, LLVM::Module, Bool, Bool>:LLVM::Function +1889
[18053045] *Crystal::CodeGenVisitor#target_def_fun<Crystal::Def+, Crystal::Type+>:LLVM::Function +213
[18153909] *Crystal::CodeGenVisitor#codegen_call<Crystal::Call, Crystal::Def+, Crystal::Type+, Array(LLVM::Value)>:(Bool | LLVM::Value | Nil) +133
[18102582] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +614
[17960447] *Crystal::ASTNode+ +1055
[17960632] *Crystal::ASTNode+ +1240
[18028129] *Crystal::CodeGenVisitor#codegen_fun<String, Crystal::Def+, Crystal::Type+, Bool, LLVM::Module, Bool, Bool>:LLVM::Function +1889
[18053045] *Crystal::CodeGenVisitor#target_def_fun<Crystal::Def+, Crystal::Type+>:LLVM::Function +213
[18153909] *Crystal::CodeGenVisitor#codegen_call<Crystal::Call, Crystal::Def+, Crystal::Type+, Array(LLVM::Value)>:(Bool | LLVM::Value | Nil) +133
[18102582] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +614
[18107968] *Crystal::CodeGenVisitor#codegen_dispatch<Crystal::Call, Array(Crystal::Def+)>:Bool +4144
[18102039] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +71
[17960447] *Crystal::ASTNode+ +1055
[18287474] *Crystal::CodeGenVisitor#visit<Crystal::Yield>:Bool +2354
[17960525] *Crystal::ASTNode+ +1133
[18287474] *Crystal::CodeGenVisitor#visit<Crystal::Yield>:Bool +2354
[17960525] *Crystal::ASTNode+ +1133
[17960632] *Crystal::ASTNode+ +1240
[18262893] *Crystal::CodeGenVisitor#visit<Crystal::While>:Bool +605
[17961707] *Crystal::ASTNode+ +2315
[17960632] *Crystal::ASTNode+ +1240
[18147248] *Crystal::CodeGenVisitor#codegen_call_with_block<Crystal::Call, Crystal::Block, Crystal::Type+, Array(LLVM::Value)>:LLVM::Value +1072
[18102605] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +637
[17960447] *Crystal::ASTNode+ +1055
[18147248] *Crystal::CodeGenVisitor#codegen_call_with_block<Crystal::Call, Crystal::Block, Crystal::Type+, Array(LLVM::Value)>:LLVM::Value +1072
[18102605] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +637
[17960447] *Crystal::ASTNode+ +1055
[18262206] *Crystal::CodeGenVisitor#codegen_if_branch<Crystal::CodeGenVisitor::Phi, Crystal::ASTNode+, LLVM::BasicBlock, Bool>:(LLVM::Value | Nil) +62
[18257560] *Crystal::CodeGenVisitor#visit<Crystal::If>:Bool +1048
[17961226] *Crystal::ASTNode+ +1834
[18262206] *Crystal::CodeGenVisitor#codegen_if_branch<Crystal::CodeGenVisitor::Phi, Crystal::ASTNode+, LLVM::BasicBlock, Bool>:(LLVM::Value | Nil) +62
[18257560] *Crystal::CodeGenVisitor#visit<Crystal::If>:Bool +1048
[17961226] *Crystal::ASTNode+ +1834
[18262206] *Crystal::CodeGenVisitor#codegen_if_branch<Crystal::CodeGenVisitor::Phi, Crystal::ASTNode+, LLVM::BasicBlock, Bool>:(LLVM::Value | Nil) +62
[18257560] *Crystal::CodeGenVisitor#visit<Crystal::If>:Bool +1048
[17961226] *Crystal::ASTNode+ +1834
[18028129] *Crystal::CodeGenVisitor#codegen_fun<String, Crystal::Def+, Crystal::Type+, Bool, LLVM::Module, Bool, Bool>:LLVM::Function +1889
[18053045] *Crystal::CodeGenVisitor#target_def_fun<Crystal::Def+, Crystal::Type+>:LLVM::Function +213
[18153909] *Crystal::CodeGenVisitor#codegen_call<Crystal::Call, Crystal::Def+, Crystal::Type+, Array(LLVM::Value)>:(Bool | LLVM::Value | Nil) +133
[18102582] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +614
[17960447] *Crystal::ASTNode+ +1055
[17960632] *Crystal::ASTNode+ +1240
[18028129] *Crystal::CodeGenVisitor#codegen_fun<String, Crystal::Def+, Crystal::Type+, Bool, LLVM::Module, Bool, Bool>:LLVM::Function +1889
[18053045] *Crystal::CodeGenVisitor#target_def_fun<Crystal::Def+, Crystal::Type+>:LLVM::Function +213
[18153909] *Crystal::CodeGenVisitor#codegen_call<Crystal::Call, Crystal::Def+, Crystal::Type+, Array(LLVM::Value)>:(Bool | LLVM::Value | Nil) +133
[18102582] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +614
[18107968] *Crystal::CodeGenVisitor#codegen_dispatch<Crystal::Call, Array(Crystal::Def+)>:Bool +4144
[18102039] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +71
[17960447] *Crystal::ASTNode+ +1055
[18028129] *Crystal::CodeGenVisitor#codegen_fun<String, Crystal::Def+, Crystal::Type+, Bool, LLVM::Module, Bool, Bool>:LLVM::Function +1889
[18053045] *Crystal::CodeGenVisitor#target_def_fun<Crystal::Def+, Crystal::Type+>:LLVM::Function +213
[18153909] *Crystal::CodeGenVisitor#codegen_call<Crystal::Call, Crystal::Def+, Crystal::Type+, Array(LLVM::Value)>:(Bool | LLVM::Value | Nil) +133
[18102582] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +614
[17960447] *Crystal::ASTNode+ +1055
[18287474] *Crystal::CodeGenVisitor#visit<Crystal::Yield>:Bool +2354
[17960525] *Crystal::ASTNode+ +1133
[18287474] *Crystal::CodeGenVisitor#visit<Crystal::Yield>:Bool +2354
[17960525] *Crystal::ASTNode+ +1133
[17960632] *Crystal::ASTNode+ +1240
[18147248] *Crystal::CodeGenVisitor#codegen_call_with_block<Crystal::Call, Crystal::Block, Crystal::Type+, Array(LLVM::Value)>:LLVM::Value +1072
[18102605] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +637
[17960447] *Crystal::ASTNode+ +1055
[17960632] *Crystal::ASTNode+ +1240
[18147248] *Crystal::CodeGenVisitor#codegen_call_with_block<Crystal::Call, Crystal::Block, Crystal::Type+, Array(LLVM::Value)>:LLVM::Value +1072
[18102605] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +637
[17960447] *Crystal::ASTNode+ +1055
[18028129] *Crystal::CodeGenVisitor#codegen_fun<String, Crystal::Def+, Crystal::Type+, Bool, LLVM::Module, Bool, Bool>:LLVM::Function +1889
[18053045] *Crystal::CodeGenVisitor#target_def_fun<Crystal::Def+, Crystal::Type+>:LLVM::Function +213
[18153909] *Crystal::CodeGenVisitor#codegen_call<Crystal::Call, Crystal::Def+, Crystal::Type+, Array(LLVM::Value)>:(Bool | LLVM::Value | Nil) +133
[18102582] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +614
[17960447] *Crystal::ASTNode+ +1055
[17960632] *Crystal::ASTNode+ +1240
[18028129] *Crystal::CodeGenVisitor#codegen_fun<String, Crystal::Def+, Crystal::Type+, Bool, LLVM::Module, Bool, Bool>:LLVM::Function +1889
[18053045] *Crystal::CodeGenVisitor#target_def_fun<Crystal::Def+, Crystal::Type+>:LLVM::Function +213
[18153909] *Crystal::CodeGenVisitor#codegen_call<Crystal::Call, Crystal::Def+, Crystal::Type+, Array(LLVM::Value)>:(Bool | LLVM::Value | Nil) +133
[18102582] *Crystal::CodeGenVisitor#visit<Crystal::Call>:Bool +614
[17960447] *Crystal::ASTNode+ +1055
[17960632] *Crystal::ASTNode+ +1240
[17645197] *Crystal::Compiler#codegen<Crystal::Program, Crystal::ASTNode+, Array(Crystal::Compiler::Source), String>:(Array(String) | Nil) +1325
[11614546] *Crystal::Command#build:Crystal::Compiler::Result +162
[9119879] *Crystal::Command#run:(Array(Crystal::ImplementationTrace) | Array(Crystal::Init::View+:Class) | Array(String) | Bool | Crystal::Compiler::Result | Hash(String, String) | IO::FileDescriptor | Nil) +1591
[8688996] main +26212
[140146440751152] __libc_start_main +240
[8531201] ???

Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
```

**Comment by asterite** [comment]
It's already fixed in master: #3377
